src/ucp/tag/tag_match.inl

```diff
@@ -212,29 +213,105 @@ ucp_tag_unexp_search(ucp_tag_match_t *tm, ucp_tag_t tag, uint64_t tag_mask,
     return NULL;
 }
 
+static UCS_F_ALWAYS_INLINE ucs_status_t
+ucp_request_recv_offload_data(ucp_request_t *req, const void *data,
+                              size_t length, unsigned recv_flags)
+{
+    ucs_status_t status = UCS_OK;
+    size_t offset;
+    ucp_offload_last_ssend_hdr_t *priv;
+
+    /* Should be used in multi-fragmented flow only */
+    ucs_assert(!(recv_flags & UCP_RECV_DESC_FLAG_EAGER_ONLY));
+
+    if (ucs_test_all_flags(recv_flags, UCP_RECV_DESC_FLAG_EAGER_LAST |
+                                       UCP_RECV_DESC_FLAG_EAGER_SYNC)) {
+        priv = (ucp_offload_last_ssend_hdr_t*)UCS_PTR_BYTE_OFFSET(data,
+                                                                  -sizeof(*priv));
+        ucp_tag_offload_sync_send_ack(req->recv.worker, priv->ssend_ack.ep_ptr,
+                                      priv->ssend_ack.sender_tag, recv_flags);
+    }
+
+    if (ucs_unlikely(req->status != UCS_OK)) {
+       return req->status;
+    }
+
+    /* There is no correct offset in middle headers with tag offload flow.
+     * All fragments are always in order - can calculate offset by
+     * subtraction of already received data.
+     * NOTE: total length of unexpected eager offload message is not known
+     * until last fragment arrives, so it is initialized to SIZE_MAX. */
+    offset = SIZE_MAX - req->recv.tag.remaining;
```

Comment:
where is req->recv.tag.remaining initialized?

Reply:
[here](https://github.com/openucx/ucx/blob/master/src/ucp/tag/eager_rcv.c#L110), and `total_length` is set to `SIZE_MAX` in case of tag offload